python/kvikio/tests/test_defaults.py

```diff
@@ -84,3 +84,34 @@ def test_bounce_buffer_size():
         kvikio.defaults.bounce_buffer_size_reset(0)
     with pytest.raises(OverflowError, match="negative value"):
         kvikio.defaults.bounce_buffer_size_reset(-1)
+
+
+def test_http_max_attempts():
+    before = kvikio.defaults.http_max_attempts()
+
+    with kvikio.defaults.set_http_max_attempts(5):
+        assert kvikio.defaults.http_max_attempts() == 5
+        kvikio.defaults.http_max_attempts_reset(4)
+        assert kvikio.defaults.http_max_attempts() == 4
+    assert before == kvikio.defaults.http_max_attempts()
```

Comment:
PyTest's errors are sometimes a bit clearer if you follow the order of `assert actual == expected`.
```suggestion
    assert kvikio.defaults.http_max_attempts() == before
```